### ISSUE TITLE
Enhance SC2178/SC2128 errors when the variable is an internal shell array variable

### ISSUE THREAD

**Issue by mat813** [koalaman/shellcheck#1101]
#### For bugs
- Rule Id (if any, e.g. SC1000): SC2178/SC2128
- My shellcheck version (`shellcheck --version` or "online"): online
- [X] I tried on shellcheck.net and verified that this is still a problem on the latest commit
- [ ] It's not reproducible on shellcheck.net, but I think that's because it's an OS, configuration or encoding issue

#### For new checks and feature suggestions
- [X] shellcheck.net (i.e. the latest commit) currently gives no useful warnings about this
- [X] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

#### Here's a snippet or screenshot that shows the problem:

```sh
#!/bin/sh

GROUPS=$1

if [ -n "${GROUPS}" ]; then
  echo "${GROUPS}";
fi
```

#### Here's what shellcheck currently says:

```
Line 3:
GROUPS=$1
^-- SC2178: Variable was used as an array but is now assigned a string.
 
Line 5:
if [ -n "${GROUPS}" ]; then
         ^-- SC2128: Expanding an array without an index only gives the first element.
 
Line 6:
  echo "${GROUPS}";
        ^-- SC2128: Expanding an array without an index only gives the first element.
```

#### Here's what I wanted or expected to see:

Something telling me `GROUPS` is an internal shell variable that is an array, which is why it's telling me this.

It took me a very long time to see that if I renamed the variable, then it stopped badgering me about it, which I was very, very puzzled about. Mostly because FreeBSD's sh, which is the one I am working with, does not have arrays at all.
it took me even longer, mostly by accident, to find out that while FreeBSD's sh did not have an internal variable named GROUPS, bash had one, and it was probably why I was getting the error.

**Comment by kapsh** [comment]
I also have a worst test-case with $GROUPS which puzzled me recently. Shellcheck doesn't warn about reusing builtin variables leading to completely wrong work.

```
❯ cat groups.sh 
#!/bin/bash

GROUPS=( hurr durr )
for g in "${GROUPS[@]}"; do
    echo "$g"
done

/tmp
❯ shellcheck groups.sh; echo $?
0

/tmp
❯ bash groups.sh 
2000
6
10
```
